### PR TITLE
Add follow mouse mode to avoid refocusing under cursor

### PR DESCRIPTION
**Pull request by outfoxxed**
#### Describe your PR, what does it fix/add?
This PR adds a new follow_mouse mode (4), that is the same as 1 (focus follows mouse) except it requires crossing a window border to re-focus the hovered window. I find it useful because my window focus won't change if my mouse gets bumped.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)
I'm fairly sure adding `m_pLastMouseFocus` at the end of `CInputManager` keeps the plugin abi the same as long as it wasn't accessing private fields, but I'm also fairly sure thats not a concern anyway.

#### Is it ready for merging, or does it need work?
Ready to merge.

### CODE REVIEW

**Comment by vaxerski** [comment]
witnof

this shouldn't be a follow_mouse opt, rather a separate on/off switch

furthermore I don't see how this changes anything

**Comment by outfoxxed** [comment]
I can change it to a separate switch.

The difference from follow_mouse=1 is that the mouse can't refocus the hovered window if the focus has been moved by the keyboard.


https://user-images.githubusercontent.com/83010835/233858565-de997335-bc14-4c9b-85c1-6e458f83319c.mp4

**Comment by vaxerski** [comment]
oh that thing

ok, yeah, bool, not mode

**Comment by outfoxxed** [comment]
I tested all the modes and it would have no effect on follow_mouse=0,2,3 because those modes don't actively follow the mouse like follow_mouse=1 does. Should it still be a bool?

**Comment by vaxerski** [comment]
yes

**Comment by outfoxxed** [comment]
`follow_mouse=4` is now `mouse_refocus=0` combined with `follow_mouse=1`.

**Comment by vaxerski** [comment]
yes, as usual. wiki pr plox

File: src/managers/input/InputManager.hpp [hunk 1]
```diff
@@ -175,6 +175,9 @@ class CInputManager {
     // for hiding cursor on touch
     bool m_bLastInputTouch = false;
 
+    // for follow mouse 4
```
Comment (vaxerski) [inline src/managers/input/InputManager.hpp:178]: comment bug

Comment (outfoxxed) [reply src/managers/input/InputManager.hpp]: Fixed

**Comment by outfoxxed** [comment]
Wiki pr above

**Review by vaxerski — APPROVED** [review]
lgtm

**Pull request merged**